### PR TITLE
Support REST in Fail Job command

### LINKED ISSUE #21377: Support the Fail Job API over REST

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

Support rest for Fail Job command

## Related issues

closes #21377

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java [hunk 1]
```diff
@@ -32,78 +35,137 @@
...
-    builder = FailJobRequest.newBuilder();
-    builder.setJobKey(key);
+    grpcRequestObjectBuilder = FailJobRequest.newBuilder();
+    grpcRequestObjectBuilder.setJobKey(key);
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+    httpRequestObject = new JobFailRequest();
+    useRest = preferRestOverGrpc;
+    jobKey = key;
+    this.jsonMapper = jsonMapper;
```
Comment (koevskinikola) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java:73]: 🔧 Why do we need a second property for the `jsonMapper`? The base class already provides it through the `objectMapper` property.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java [hunk 2]
```diff
@@ -32,78 +35,137 @@
...
   public FailJobCommandStep2 setVariablesInternal(final String variables) {
-    builder.setVariables(variables);
+    grpcRequestObjectBuilder.setVariables(variables);
+    // This check is mandatory. Without it, gRPC requests can fail unnecessarily.
+    // gRPC and REST handle setting variables differently:
+    // - For gRPC commands, we only check if the JSON is valid and forward it to the engine.
+    //    The engine checks if the provided String can be transformed into a Map, if not it
+    //    throws an error.
+    // - For REST commands, users have to provide a valid JSON Object String.
+    //    Otherwise, the client throws an exception already.
```
Comment (koevskinikola) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java:106]: 👍 Great explanation!

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/FailJobTest.java
Comment (koevskinikola) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/FailJobTest.java:1]: 👍 Awesome! You converted it to a JUnit 5 test.

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Looks great @nicpuppa! 🎉 

I only have a minor thing to clarify and we can approve and merge this PR.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java (hunk 1 above)
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/FailJobCommandImpl.java:73]: Good catch 👍

**Review by koevskinikola — APPROVED** [review]
🚀

**Pull request merged**